modules/shared/src/main/scala/org/tesselation/ext/codecs.scala

```diff
@@ -0,0 +1,34 @@
+package org.tesselation.ext
+
+import cats.effect.Concurrent
+import cats.syntax.option._
+
+import scala.reflect.ClassTag
+
+import org.tesselation.kryo.KryoSerializer
+
+import org.http4s.DecodeResult.{failureT, successT}
+import org.http4s.{EntityDecoder, EntityEncoder, MalformedMessageBodyFailure}
+
+object codecs {
+
+  object BinaryCodec {
+    implicit def encoder[F[_]: KryoSerializer, A <: AnyRef]: EntityEncoder[F, A] =
+      EntityEncoder.byteArrayEncoder[F].contramap { anyRef =>
+        KryoSerializer[F].serialize(anyRef) match {
+          case Right(bytes) => bytes
+          case Left(ex)     => throw ex // FIXME: how to solve this?
```

Comment:
Have you checked how it works? Will it throw in the runtime during encoding process? If so, it's imo fine. Kryo throws in the runtime and we can do nothing with that actually
tl;dr; as long as the error is not swallowed, it's imo fine